0.4 Release (#20)
ocwg/spec: spec/0.4/extensions.md

# OCIF Extensions

> **OCWG Working Draft, 21 January 2025**

Reviewer: ```suggestion
```

Perhaps we remove this?

> Copyright © 2025 the Contributors to the Open Canvas Working Group (OCWG). 

Reviewer: There's a "legal" section a few lines down that's more specific.
```suggestion
```

> Versioning Policy: Currently, all extensions in this document are at version **0.3**.

Reviewer: ```suggestion
Versioning Policy: Currently, all extensions in this document are at version **0.4**.
```

## Abstract
This document describes extensions to the Open Canvas Interchange Format (OCIF) that are not part of the core specification. Extensions are optional and may be used to add additional functionality to OCIF documents.
See the [OCIF Core Specification](./spec) for the core features of OCIF.

## Status of this Document
This document is an editor's draft and has no official standing. It is a work in progress and may be updated, replaced, or obsoleted by other documents at any time.

**Legal**:

> Open Canvas Interchange Format (OCIF) v0.2 © 2024 by Open Canvas Working Group is licensed under CC BY-SA 4.0. To view a copy of this licence, visit https://creativecommons.org/licenses/by-sa/4.0/

Reviewer: ```suggestion
Open Canvas Interchange Format (OCIF) v0.4 © 2025 by Open Canvas Working Group is licensed under CC BY-SA 4.0. To view a copy of this licence, visit https://creativecommons.org/licenses/by-sa/4.0/
```

### Table of Contents
<!-- TOC -->
- [OCIF Extensions](#ocif-extensions)
  - [Abstract](#abstract)
  - [Status of this Document](#status-of-this-document)
    - [Table of Contents](#table-of-contents)
- [Node Extensions](#node-extensions)
  - [Ports Node](#ports-node)
  - [Relative Node](#relative-node)
- [Relation Extensions](#relation-extensions)
  - [Hyperedge Relation](#hyperedge-relation)
  - [Parent-Child Relation](#parent-child-relation)
- [Changes](#changes)
<!-- TOC -->

# Node Extensions
These are  [extension](spec.md#extensions) that can be added to nodes in an OCIF document.
To be placed inside the `data` `array`.

## Ports Node
- Name: `@ocif/node/ports`

> - URI: `https://spec.canvasprotocol.org/0.3/extensions/ports-node.json`

Reviewer: ```suggestion
- URI: `https://spec.canvasprotocol.org/0.4/extensions/ports-node.json`
```
